x-pack/plugins/ml/public/jobs/new_job/simple/recognize/check_module.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import chrome from 'ui/chrome';
+import { mlJobService } from '../../../../services/job_service';
+import { ml } from '../../../../services/ml_api_service';
+import { toastNotifications } from 'ui/notify';
+
+
+// Checks whether the jobs in a data recognizer module have been created.
+// Redirects to the Anomaly Explorer to view the jobs if they have been created,
+// or the recognizer job wizard for the module if not.
+export function checkViewOrCreateJobs(Private, $route, kbnBaseUrl, kbnUrl, i18n) {
+
+  return new Promise((resolve, reject) => {
+    const moduleId = $route.current.params.id;
+    const indexPatternId = $route.current.params.index;
+
+    // Load the module, and check if the job(s) in the module have been created.
+    // If so, load the jobs in the Anomaly Explorer.
+    // Otherwise open the data recognizer wizard for the module.
+    ml.dataRecognizerModuleJobsExist({ moduleId })
+      .then((resp) => {
+        const basePath = `${chrome.getBasePath()}/app/`;
+
+        if (resp.jobsExist === true) {
+          const resultsPageUrl = mlJobService.createResultsUrlForJobs(resp.jobs, 'explorer');
+          window.location.href = `${basePath}${resultsPageUrl}`;
+          reject();
```

Comment:
Could you clarify why we need to reject here?

Reply:
The two `reject() calls after `window.location.href` aren't strictly needed, but have left them in for completeness to make it clear the Promise is being rejected rather than resolved. Have added in comments to the code.